crates/driver/src/infra/persistence/mod.rs

```diff
@@ -0,0 +1,70 @@
+use {
+    crate::{domain::competition::auction::Id, infra::solver::Config},
+    serde::Deserialize,
+    std::sync::Arc,
+    tracing::Instrument,
+};
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
```

Comment:
A good rule of thumb is probably that nothing should ever have to be (de)serialized outside of config loading and request sending. That way the separation between domain and everything else becomes very clear. Maybe we can even have a clippy lint for that to really enforce the DDD style separation of concerns regarding this. 🤔

Reply:
@MartinquaXD I see your point. In the end it is a matter of architectural decision, There are also advantages of doing it how I did it: each module is standalone and controls all the data (config, dto and domain), and the config module would be just an aggregator of configs. As it is in the code, everything is in the module (dto and domain) but not the config. Each way has its cons/pros.

I personally like it more when each module has full control, but if all the modules are gathered like that it makes no sense for me to break that consistency. I would create chaos :ok_hand:

Reply:
I think having a clippy is a good idea, but sine there is (de)serialization also in all the `dto` modules within the domain logic, I am not so sure `clippy` can be customized so fine to detect and skip all of those :thinking: